utils/utils.go

```diff
@@ -0,0 +1,356 @@
+func APIServerVersionInfo(t *testing.T, clientset *kubernetes.Clientset) {
+	version, err := clientset.DiscoveryClient.ServerVersion()
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Printf("API Server Version: %s.%s\n", version.Major, version.Minor)
+	fmt.Printf("API Server GitVersion: %s\n", version.GitVersion)
+}
+
+// AssertEKSConfigMapReady ensures that the EKS aws-auth ConfigMap
+// exists and has data.
+func AssertEKSConfigMapReady(t *testing.T, clientset *kubernetes.Clientset) {
+	awsAuthReady, retries := false, MaxRetries
+	configMapName, namespace := "aws-auth", "kube-system"
+	var awsAuth *v1.ConfigMap
+	var err error
+
+	for !awsAuthReady && retries > 0 {
+		awsAuth, err = clientset.CoreV1().ConfigMaps(namespace).Get(configMapName, metav1.GetOptions{})
+		if err != nil {
+			fmt.Printf("Waiting for '%s' ConfigMap to be returned. Retrying...\n", configMapName)
+			retries--
+			time.Sleep(RetryInterval * time.Second)
+			continue
+		}
+		awsAuthReady = true
+		break
+	}
+
+	require.Equal(t, true, awsAuthReady, "EKS ConfigMap '%s' does not exist in namespace '%s'", configMapName, namespace)
+	require.NotNil(t, awsAuth.Data, "%s ConfigMap should not be nil", configMapName)
+	require.NotEmpty(t, awsAuth.Data, "'%s' ConfigMap should not be empty", configMapName)
+	fmt.Printf("EKS ConfigMap '%s' exists and has data\n", configMapName)
+	t.Logf("EKS ConfigMap '%s' exists and has data:\n%s", configMapName, awsAuth.Data)
+}
+
+// AssertAllNodesReady ensures that all nodes are running & have a "Ready"
+// status condition.
+func AssertAllNodesReady(t *testing.T, clientset *kubernetes.Clientset, desiredNodeCount int) {
+	var nodes *v1.NodeList
+	var err error
+
+	fmt.Printf("Total Desired Worker Node Count: %d\n", desiredNodeCount)
+
+	// Validate that the desired worker node count of instances are running.
+	retries := MaxRetries
+	for retries > 0 {
+		nodes, err = clientset.CoreV1().Nodes().List(metav1.ListOptions{})
+		if err != nil || desiredNodeCount != len(nodes.Items) {
+			fmt.Printf("Waiting for a desired worker Node count of (%d) instances to be running. Retrying...\n", desiredNodeCount)
+			retries--
+			time.Sleep(RetryInterval * time.Second)
+		} else {
+			break
+		}
+	}
+
+	require.NotEmpty(t, nodes, "The Nodes list returned should not be empty")
+	require.Equal(t, desiredNodeCount, len(nodes.Items),
+		"%d out of %d desired worker nodes are instantiated and running", len(nodes.Items), desiredNodeCount)
+
+	// Validate each node has a "Ready" status.
+	i, readyCount := 0, 0
+	nodeReady, retries := false, MaxRetries
+	for i < len(nodes.Items) && retries > 0 {
+		node := nodes.Items[i]
+		n, err := clientset.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
+		if err != nil {
+			fmt.Printf("Waiting for Node '%s' to be returned. Retrying...\n", node.Name)
+			retries--
+			time.Sleep(RetryInterval * time.Second)
+			continue
+		}
+		for _, condition := range n.Status.Conditions {
+			if condition.Type == "Ready" {
```

Comment:
Prefer using the defined condition consts here: https://github.com/kubernetes/api/blob/db5a9d1c40ebe3cdff47aab1b7ecdcd22f841ef8/core/v1/types.go#L2412-L2424

Reply:
Done